### DISCUSSION TITLE
how to delete a widget

### DISCUSSION THREAD

**Discussion by plugeit** [flet-dev/flet#644] [Q&A] [answered]
please answer and also any way to delete all the items

**Comment by ndonkoHenri** [comment, marked as answer]
If you mean - how to remove a control from the `Page`, then simply remove it from its controls.
`page.controls` is a list, so you could use:
- `page.controls.clear()` to remove all the items/controls
- `page.controls.remove(x)` to remove a particular control. The x parameter refers to the item/control to be removed from the list. If you stored to control to be deleted in a variable, then set item to that variable.
- `page.controls.pop()` to remove the last control/item in the list. You could pass an index parameter to it which is the control's index in the list.

See [this](https://www.edureka.co/blog/python-list-remove/amp/) if you don't still understand.

All the above methods (pop, delete, remove)also applies for Column, Row, and all the others. That's just python. 😉

**Reply by plugeit** [reply]
thanks bro you are the savior who always saves me

**Reply by ndonkoHenri** [reply]
You could mark my response as an answer so those viewing could see it.
